src/Database.ts

```diff
@@ -56,15 +56,15 @@ export class Database<
       if (mode) {
         this.db = new driver(filename, mode, err => {
           if (err) {
-            return reject(err)
+            return reject(new Error(err)) // Wrapping to build a stack trace.
```

Comment:
You should do a typecheck

```
if (err instanceof Error) {
   return reject(new Error(err.message))
}

return reject(new Error(err))
```

Reply:
I don't recommend using your suggested code. `new Error(err)` expects err to already be an 'Error' object. If its a string, that's fine too, but it looks like the errors returned have a `message` and an `errno` and a `code`. `new Error(err)` will retain all those additional properties, while `new Error(err.message)` obviously won't. And just because err is an instanceof Error doesn't mean it has a proper stack trace.

Reply:
I have never heard of error taking in the properties of another error other than the message when fed. I tried it in the console as that would be nice if it did, but it doesn't:

<img width="287" alt="image" src="https://user-images.githubusercontent.com/855434/156988180-1097d38e-538a-4730-bd84-a7f856ebce12.png">

Reply:
Hmm, I could have sworn I tested that, but I must have done something weird because I can confirme you're right. In any case, that can be replaced with:

```
      let newError = new Error(e)
      for(let k in e) {
        newError[k] = e[k]
      }
      throw newError
```